### PR TITLE
Updating Audit Log Configurations

### PR DESCRIPTION
**Pull request by greg-szrama**
Resolves: https://github.com/redpanda-data/documentation-private/issues/2153

Added audit log items to cluster properties, updated two property defaults, and added documentation for new option.

[Updated audit log configuration descriptions.](https://deploy-preview-201--redpanda-docs-preview.netlify.app/23.3/manage/audit-logging/#audit-log-configuration-options)

[Added to cluster properties](https://deploy-preview-201--redpanda-docs-preview.netlify.app/23.3/reference/cluster-properties/#audit-logging-properties)

### CODE REVIEW

File: modules/manage/pages/audit-logging.adoc [hunk 1]
```diff
@@ -23,14 +23,15 @@ Messages recorded to the audit log topic comply with the https://schema.ocsf.io/
...
-* `audit_enabled_event_types`: List of strings in JSON style identifying the event types to include in the audit log. This may include any of the following - `management, produce, consume, describe, heartbeat, authenticate, schema_registry, admin`. Default: `'["management","authenticate","admin"]'`.
-* `audit_exclude_topics`: List of strings in JSON style identifying the topics the audit logging system should ignore. This list cannot include the `__audit_log` topic. Redpanda will reject the command if you do attempt to include that topic. Default: null.
-* `audit_exclude_principals`: List of strings in JSON style identifying the principals the audit logging system should ignore. Principals can be listed as `User:name` or `name`, both are accepted. Default: null.
+* xref:reference:cluster-properties.adoc#audit_enabled[`audit_enabled`]: Boolean value to enable audit logging. When you set this to true, Redpanda checks for an existing topic named `__audit_log`. If none is found, Redpanda automatically creates one for you. Default: `false`.
```
Comment (michael-redpanda) [inline modules/manage/pages/audit-logging.adoc]: FYI, there was a last minute request from Alex to adjust the naming of certain internal topics, which included the audit log (PR: https://github.com/redpanda-data/redpanda/pull/15817).

So the name of the audit log is no longer `__audit_log` it is `_redpanda.audit_log`

File: modules/manage/pages/audit-logging.adoc [hunk 2]
```diff
@@ -23,14 +23,15 @@ Messages recorded to the audit log topic comply with the https://schema.ocsf.io/
...
+* xref:reference:cluster-properties.adoc#audit_log_num_partitions[`audit_log_num_partitions`]: Integer value defining the number of partitions used by a newly created audit topic. This configuration applies only to the audit log topic and may be different from the cluster or other topic configurations. This cannot be altered for existing audit log topics. Default: `12`.
+* xref:reference:cluster-properties.adoc#audit_log_replication_factor[`audit_log_replication_factor`]: Integer value defining the replication factor for a newly created audit log topic. This configuration applies only to the audit log topic and may be different from the cluster or other topic configurations. This cannot be altered for existing audit log topics. Default: `3`.
```
Comment (michael-redpanda) [inline modules/manage/pages/audit-logging.adoc]:
```suggestion
* xref:reference:cluster-properties.adoc#audit_log_replication_factor[`audit_log_replication_factor`]: Optional integer value defining the replication factor for a newly created audit log topic. This configuration applies only to the audit log topic and may be different from the cluster or other topic configurations. This cannot be altered for existing audit log topics. If a value is not provided, will use the `internal_topic_replication_factor` cluster config value.  Default: `nill`.
```

File: modules/manage/pages/audit-logging.adoc [hunk 3]
```diff
@@ -23,14 +23,15 @@ Messages recorded to the audit log topic comply with the https://schema.ocsf.io/
...
+* xref:reference:cluster-properties.adoc#audit_enabled_event_types[`audit_enabled_event_types`]: List of strings in JSON style identifying the event types to include in the audit log. This may include any of the following - `management, produce, consume, describe, heartbeat, authenticate, schema_registry, admin`. Default: `'["management","authenticate","admin"]'`.
+* xref:reference:cluster-properties.adoc#audit_exclude_topics[`audit_exclude_topics`]: List of strings in JSON style identifying the topics the audit logging system should ignore. This list cannot include the `__audit_log` topic. Redpanda will reject the command if you do attempt to include that topic. Default: null.
+* xref:reference:cluster-properties.adoc#audit_queue_drain_interval[`audit_queue_drain_interval`]: Internally, Redpanda batches audit log messages in memory and periodically writes them to the audit log topic. This defines the period in milliseconds between draining this queue to the audit log topic. Longer intervals may help prevent duplicate messages, especially in high throughput scenarios, but they also increase the risk of data loss during hard shutdowns where the queue is lost. Default: `500`.
```
Comment (michael-redpanda) [inline modules/manage/pages/audit-logging.adoc]:
```suggestion
* xref:reference:cluster-properties.adoc#audit_queue_drain_interval_ms[`audit_queue_drain_interval_ms`]: Internally, Redpanda batches audit log messages in memory and periodically writes them to the audit log topic. This defines the period in milliseconds between draining this queue to the audit log topic. Longer intervals may help prevent duplicate messages, especially in high throughput scenarios, but they also increase the risk of data loss during hard shutdowns where the queue is lost. Default: `500`.
```

File: modules/manage/pages/audit-logging.adoc [hunk 4]
```diff
@@ -23,14 +23,15 @@ Messages recorded to the audit log topic comply with the https://schema.ocsf.io/
...
+* xref:reference:cluster-properties.adoc#audit_queue_max_buffer_size_per_shard[`audit_queue_max_buffer_size_per_shard`]: Integer value defining the maximum amount of memory in bytes used by the audit buffer in each shard. Once this size is reached, requests to log additional audit messages will return a non-retryable error. You must restart the cluster when changing this value. Default: `1048576`.
+* xref:reference:cluster-properties.adoc#audit_enabled_event_types[`audit_enabled_event_types`]: List of strings in JSON style identifying the event types to include in the audit log. This may include any of the following - `management, produce, consume, describe, heartbeat, authenticate, schema_registry, admin`. Default: `'["management","authenticate","admin"]'`.
+* xref:reference:cluster-properties.adoc#audit_exclude_topics[`audit_exclude_topics`]: List of strings in JSON style identifying the topics the audit logging system should ignore. This list cannot include the `__audit_log` topic. Redpanda will reject the command if you do attempt to include that topic. Default: null.
```
Comment (michael-redpanda) [inline modules/manage/pages/audit-logging.adoc]:
```suggestion
* xref:reference:cluster-properties.adoc#audit_exclude_topics[`audit_exclude_topics`]: List of strings in JSON style identifying the topics the audit logging system should ignore. This list cannot include the `_redpanda.audit_log` topic. Redpanda will reject the command if you do attempt to include that topic. Default: null.
```

File: modules/manage/pages/audit-logging.adoc (hunk 1 above)
Comment (michael-redpanda) [inline modules/manage/pages/audit-logging.adoc]:
```suggestion
* xref:reference:cluster-properties.adoc#audit_enabled[`audit_enabled`]: Boolean value to enable audit logging. When you set this to true, Redpanda checks for an existing topic named `_redpanda.audit_log`. If none is found, Redpanda automatically creates one for you. Default: `false`.
```

File: modules/reference/pages/cluster-properties.adoc [hunk 5]
```diff
@@ -29,6 +29,107 @@ List of superuser usernames.
...
+Defines the replication factor for a newly created audit log topic. This configuration applies only to the audit log topic and may be different from the cluster or other topic configurations. This cannot be altered for existing audit log topics.
+
+*Default*: `3`
+
+*Restart required*: no
+
+---
+
+[[audit_queue_drain_interval]]
+=== audit_queue_drain_interval
```
Comment (michael-redpanda) [inline modules/reference/pages/cluster-properties.adoc]:
```suggestion
[[audit_queue_drain_interval_ms]]
=== audit_queue_drain_interval_ms
```

File: modules/reference/pages/cluster-properties.adoc [hunk 6]
```diff
@@ -29,6 +29,107 @@ List of superuser usernames.
...
+---
+
+[[audit_log_replication_factor]]
+=== audit_log_replication_factor
+
+Defines the replication factor for a newly created audit log topic. This configuration applies only to the audit log topic and may be different from the cluster or other topic configurations. This cannot be altered for existing audit log topics.
+
+*Default*: `3`
+
+*Restart required*: no
```
Comment (michael-redpanda) [inline modules/reference/pages/cluster-properties.adoc]:
```suggestion
[[audit_log_replication_factor]]
=== audit_log_replication_factor

Defines the replication factor for a newly created audit log topic. This configuration applies only to the audit log topic and may be different from the cluster or other topic configurations. This cannot be altered for existing audit log topics.  This value is optional.  If it is not provided, the value for `internal_topic_replication_factor` will be used.

*Default*: `nill`

*Restart required*: no
```

File: modules/reference/pages/cluster-properties.adoc [hunk 7]
```diff
@@ -29,6 +29,107 @@ List of superuser usernames.
 
 ---
 
+== Audit logging properties
+
+[[audit_enabled]]
+=== audit_enabled
+
+Enables or disables audit logging. When you set this to true, Redpanda checks for an existing topic named `__audit_log`. If none is found, Redpanda automatically creates one for you.
```
Comment (michael-redpanda) [inline modules/reference/pages/cluster-properties.adoc]:
```suggestion
Enables or disables audit logging. When you set this to true, Redpanda checks for an existing topic named `_redpanda.audit_log`. If none is found, Redpanda automatically creates one for you.
```

File: modules/reference/pages/cluster-properties.adoc [hunk 8]
```diff
@@ -29,6 +29,107 @@ List of superuser usernames.
...
+*Default*: null
+
+*Restart required*: no
+
+---
+
+[[audit_exclude_topics]]
+=== audit_exclude_topics
+
+List of strings in JSON style identifying the topics the audit logging system should ignore. This list cannot include the `__audit_log` topic. Redpanda will reject the command if you do attempt to include that topic.
```
Comment (michael-redpanda) [inline modules/reference/pages/cluster-properties.adoc]:
```suggestion
List of strings in JSON style identifying the topics the audit logging system should ignore. This list cannot include the `_redpanda.audit_log` topic. Redpanda will reject the command if you do attempt to include that topic.
```

**Review by michael-redpanda — COMMENTED** [review]
Thank you!

File: modules/manage/pages/audit-logging.adoc [hunk 9]
```diff
@@ -23,14 +23,15 @@ Messages recorded to the audit log topic comply with the https://schema.ocsf.io/
...
+* xref:reference:cluster-properties.adoc#audit_queue_drain_interval[`audit_queue_drain_interval`]: Internally, Redpanda batches audit log messages in memory and periodically writes them to the audit log topic. This defines the period in milliseconds between draining this queue to the audit log topic. Longer intervals may help prevent duplicate messages, especially in high throughput scenarios, but they also increase the risk of data loss during hard shutdowns where the queue is lost. Default: `500`.
+* xref:reference:cluster-properties.adoc#audit_exclude_principals[`audit_exclude_principals`]: List of strings in JSON style identifying the principals the audit logging system should ignore. Principals can be listed as `User:name` or `name`, both are accepted. Default: null.
 
 Even though audited event messages are stored to a specialized immutable topic, standard topic settings still apply. For example, you can apply the same Tiered Storage, retention time, and replication settings available to normal topics. These particular options are important for controlling the amount of disk space utilized by your audit topics.
```
Comment (JakeSCahill) [inline modules/manage/pages/audit-logging.adoc:36]: At the bottom of this page is a leftover placeholder `SPECIFY AN EXCESSIVELY LONG VALUE HERE`.

Comment (greg-szrama) [reply modules/manage/pages/audit-logging.adoc:36]: I have no clue how that got through review. Thanks for catching that!

**Comment by michael-redpanda** [comment]
@greg-szrama thanks for the updates are there any other references to `__audit_log` in the docs that aren't covered by this PR?

**Comment by greg-szrama** [comment]
> @greg-szrama thanks for the updates are there any other references to `__audit_log` in the docs that aren't covered by this PR?

I double-checked the entire docs project, I got them all corrected under this PR!

**Review by deniscoady — APPROVED** [review]

**Review by michael-redpanda — APPROVED** [review]
lgmt thank you @greg-szrama for the quick turn around

**Review by JakeSCahill — APPROVED** [review]

**Pull request merged**